src/lib.rs

```diff
@@ -115,3 +121,90 @@ pub enum Error {
 /// Result of an high level computation such as in Alice and Bob roles executing the protocol,
 /// wraps the crate level [`enum@Error`] type.
 pub type Res<T> = Result<T, Error>;
+
+/// A unique identifier used to identify trades and swaps.
+///
+/// This is a wrapper against `uuid::Uuid` with `StrictEncode` and `StrictDecode` implementation.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, Display, Serialize, Deserialize)]
+#[display(inner)]
+pub struct Uuid(uuid::Uuid);
+
+impl Uuid {
+    /// Creates a new uuid for trades and swaps.
+    pub fn new() -> Self {
+        Self(uuid::Uuid::new_v4())
+    }
+
+    /// Creates a new random uuid, same as `Self::new()`.
```

Comment:
What is the purpose of this extra function?

Reply:
To keep the API compatible with what we have before: `SwapId::random()`